From 5410c2c8970bf599806dd3d6d63a8870b4a78905 Mon Sep 17 00:00:00 2001
From: Ulf Samuelsson <ulf.samuelsson@atmel.com>
Date: Sat, 16 Apr 2011 02:32:46 +0200
Subject: [PATCH 19/24] at91: introduce at91sam9m10g45ek board

Slight differences oblige us to create a new board and mach-id for this
at91sam9m10g45 Evaluation Kit. Introduce the basic board file changes.

Signed-off-by: Nicolas Ferre <nicolas.ferre@atmel.com>

Ported to linux-2.6.38
Signed-off-by: Ulf Samuelsson <ulf.samuelsson@atmel.com>
---
 arch/arm/mach-at91/Kconfig              |   19 +++++++++++--------
 arch/arm/mach-at91/Makefile             |    5 ++---
 arch/arm/mach-at91/board-sam9m10g45ek.c |   13 +++++++++----
 3 files changed, 22 insertions(+), 15 deletions(-)

diff --git a/arch/arm/mach-at91/Kconfig b/arch/arm/mach-at91/Kconfig
index 53c1379..65f59c7 100644
--- a/arch/arm/mach-at91/Kconfig
+++ b/arch/arm/mach-at91/Kconfig
@@ -518,23 +518,26 @@ if ARCH_AT91SAM9M10
 
 comment "AT91SAM9M10 Board Type"
 
-config MACH_AT91SAM9M10G45EK
-	bool "Atmel AT91SAM9M10G45-EK Evaluation Kits"
-	select HAVE_NAND_ATMEL_BUSWIDTH_16
-	help
-	  Select this if you are using Atmel's AT91SAM9M10G45-EK Evaluation Kit.
-	  "ES" at the end of the name means that this board is an
-	  Engineering Sample.
+choice 
+	prompt "Atmel Board"
 
 config MACH_AT91SAM9M10EKES
         bool "Atmel AT91SAM9M10-EKES Evaluation Kit"
 	select HAVE_NAND_ATMEL_BUSWIDTH_16
-        depends on ARCH_AT91SAM9M10
         help
           Select this if you are using Atmel's AT91SAM9M10-EKES Evaluation Kit.
           "ES" at the end of the name means that this board is an
           Engineering Sample.
 
+config MACH_AT91SAM9M10G45EK
+        bool "Atmel AT91SAM9M10G45-EK Evaluation Kit"
+	select HAVE_NAND_ATMEL_BUSWIDTH_16
+        help
+          Select this if you are using Atmel's AT91SAM9M10G45-EK Evaluation Kit.
+	  <http://atmel.com/dyn/products/tools_card_v2.asp?tool_id=4735>
+
+endchoice
+
 endif
 
 # ----------------------------------------------------------
diff --git a/arch/arm/mach-at91/Makefile b/arch/arm/mach-at91/Makefile
index b4d6cec..43fdf3c 100644
--- a/arch/arm/mach-at91/Makefile
+++ b/arch/arm/mach-at91/Makefile
@@ -75,10 +75,9 @@ obj-$(CONFIG_MACH_GSIA18S)	+= board-gsia18s.o board-stamp9g20.o
 obj-$(CONFIG_MACH_SNAPPER_9260)	+= board-snapper9260.o
 
 # AT91SAM9G45 board-specific support
-obj-$(CONFIG_MACH_AT91SAM9M10G45EK) += board-sam9m10g45ek.o
-
-# AT91SAM9M10 board-specific support
+obj-$(CONFIG_MACH_AT91SAM9G45EKES) += board-sam9m10g45ek.o
 obj-$(CONFIG_MACH_AT91SAM9M10EKES) += board-sam9m10g45ek.o
+obj-$(CONFIG_MACH_AT91SAM9M10G45EK) += board-sam9m10g45ek.o
 
 # AT91CAP9 board-specific support
 obj-$(CONFIG_MACH_AT91CAP9ADK)	+= board-cap9adk.o
diff --git a/arch/arm/mach-at91/board-sam9m10g45ek.c b/arch/arm/mach-at91/board-sam9m10g45ek.c
index 9859011..e2ad65a 100644
--- a/arch/arm/mach-at91/board-sam9m10g45ek.c
+++ b/arch/arm/mach-at91/board-sam9m10g45ek.c
@@ -2,6 +2,7 @@
  *  Board-specific setup code for the AT91SAM9M10G45 Evaluation Kit family
  *
  *  Covers: * AT91SAM9G45-EKES  board
+ *          * AT91SAM9M10-EKES  board
  *          * AT91SAM9M10G45-EK board
  *
  *  Copyright (C) 2009 Atmel Corporation.
@@ -251,9 +252,14 @@ static struct atmel_lcdfb_info __initdata ek_lcdc_data;
  * Touchscreen
  */
 static struct at91_tsadcc_data ek_tsadcc_data = {
+#if defined(CONFIG_MACH_AT91SAM9M10G45EK)
+	.adc_clock		= 800000,
+	.ts_sample_hold_time	= 0x04,
+#else
 	.adc_clock		= 300000,
-	.pendet_debounce	= 0x0d,
 	.ts_sample_hold_time	= 0x0a,
+#endif
+	.pendet_debounce	= 0x0d,
 };
 
 
@@ -422,13 +428,12 @@ static void __init ek_board_init(void)
 	at91_pwm_leds(ek_pwm_led, ARRAY_SIZE(ek_pwm_led));
 }
 
-
 #if defined(CONFIG_MACH_AT91SAM9G45EKES)
 MACHINE_START(AT91SAM9G45EKES, "Atmel AT91SAM9G45-EKES")
-#elif defined(CONFIG_MACH_AT91SAM9M10G45EK)
-MACHINE_START(AT91SAM9M10G45EK, "Atmel AT91SAM9M10G45-EK")
 #elif defined(CONFIG_MACH_AT91SAM9M10EKES)
 MACHINE_START(AT91SAM9M10EKES, "Atmel AT91SAM9M10-EKES")
+#elif defined(CONFIG_MACH_AT91SAM9M10EKES)
+MACHINE_START(AT91SAM9M10G45EK, "Atmel AT91SAM9M10G45-EK")
 #endif
 	/* Maintainer: Atmel */
 	.boot_params	= AT91_SDRAM_BASE + 0x100,
-- 
1.6.3.3

